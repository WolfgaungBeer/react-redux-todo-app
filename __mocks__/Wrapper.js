import React from 'react';
import { node } from 'prop-types';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; // eslint-disable-line
import { Theme, defaultTheme } from '../src/components';

const mockStore = configureStore();
const store = mockStore({ scado: { theme: defaultTheme } });

const propTypes = {
    children: node,
};

const defaultProps = {
    children: undefined,
};

const Wrapper = ({ children }) => (
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

Wrapper.propTypes = propTypes;
Wrapper.defaultProps = defaultProps;

export default Wrapper;
