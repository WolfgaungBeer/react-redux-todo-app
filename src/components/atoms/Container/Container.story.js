/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import Container from './';


storiesOf('Container', module)
    .add('default', () => (
        <Container>Container</Container>
    ))
    .add('with props', () => (
        <Container
            position={text('position', 'relative')}
            top={text('top', '0px')}
            left={text('left', '0px')}
            right={text('right', '0px')}
            bottom={text('bottom', '0px')}
            width={text('width', '100%')}
            height={text('height', '100%')}
        >
            Container
        </Container>
    ));
