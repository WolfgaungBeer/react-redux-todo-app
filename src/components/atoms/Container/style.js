export const getStyle = () => (props) => {
    const { position, top, left, bottom, right, width, height, styles } = props;
    return `
        position: ${position};
        top: ${top};
        left: ${left};
        right: ${right};
        bottom: ${bottom};
        width: ${width};
        height: ${height};
        ${styles && styles(props)};
    `;
};
