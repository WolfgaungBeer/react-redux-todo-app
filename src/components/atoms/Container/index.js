import styled from 'styled-components';
import { string, func, node } from 'prop-types';
import { getStyle } from './style';

const propTypes = {
    position: string,
    top: string,
    left: string,
    right: string,
    bottom: string,
    width: string,
    height: string,
    className: string,
    styles: func,
    children: node,
};

const defaultProps = {
    position: undefined,
    top: undefined,
    left: undefined,
    right: undefined,
    bottom: undefined,
    width: undefined,
    height: undefined,
    className: undefined,
    styles: undefined,
    children: undefined,
};

const Container = styled.div`
    ${getStyle()}
`;

Container.propTypes = propTypes;
Container.defaultProps = defaultProps;

export default Container;
