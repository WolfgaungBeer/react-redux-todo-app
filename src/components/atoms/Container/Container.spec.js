import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../../../__mocks__/Wrapper';
import Container from './';

it('renders the Container Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <Container>Container</Container>
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <Container
                position="absolute"
                top="0px"
                left="0px"
                width="100%"
                height="100%"
            >
                Container
            </Container>
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Wrapper>
            <Container
                position="absolute"
                top="0px"
                left="0px"
                width="100%"
                height="100%"
                styles={() => 'background-color: red;'}
            >
                Container
            </Container>
        </Wrapper>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();
});
