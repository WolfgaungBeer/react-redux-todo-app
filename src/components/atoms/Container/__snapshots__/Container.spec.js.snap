// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Container Component correctly 1`] = `
<div
  className="sc-bdVaJa cynRIR"
  height={undefined}
  width={undefined}
>
  Container
</div>
`;

exports[`renders the Container Component correctly 2`] = `
<div
  className="sc-bdVaJa cRtKJV"
  height="100%"
  width="100%"
>
  Container
</div>
`;

exports[`renders the Container Component correctly 3`] = `
<div
  className="sc-bdVaJa ghkxGi"
  height="100%"
  width="100%"
>
  Container
</div>
`;
